### ISSUE TITLE
`systemd-vconsole-setup` & `systemd-firstboot` don't handle `console=ttynull`

### ISSUE THREAD

**Issue by Winterhuman** [systemd/systemd#31257]
### systemd version the issue has been seen with

255

### Used distribution

Arch Linux

### Linux kernel version used

6.7.4-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

systemd

### Expected behaviour you didn't see

When `console=ttynull` is set, `systemd-vconsole-setup.service` & `systemd-firstboot.service` should gracefully exit if `/proc/consoles` is empty, which indicates the NULL TTY driver is enabled.

Note: Checking `/proc/consoles` is necessary since, supposedly, there are ways to enable the NULL TTY driver besides setting `console=ttynull`, so adding `ConditionKernelCommandLine=!console=ttynull` wouldn't be enough (hence why I didn't send a pull to simply add the condition lines; it wouldn't be sufficient).

### Unexpected behaviour you saw

Side issue: `systemd-vconsole-setup` & `systemd-firstboot` don't seem to have "component" tags when I opened this issue.

Both services fail since they're trying to open `/dev/console`, however, the NULL TTY driver discards all writes and makes using the device for stdin, stdout, and stderr impossible (see [this kernel mailing list message](https://lore.kernel.org/linux-doc/20220215005615.11244-1-rdunlap@infradead.org/T/)):

```
systemd-vconsole-setup: Failed to lock /dev/console: No such device
...
systemd-firstboot.service: Failed to set up standard input: No such device
```

### Steps to reproduce the problem

1. Boot with `console=ttynull` set on the kernel cmdline.
2. Observe errors mentioned above.

**Comment by poettering** [comment]
do we really have to support that? if am not sure we need to cover this case? sounds like a specialist case, what does using console=ttynull even get you that just turning off the status output of system wouldn#t get you?

**Comment by poettering** [comment]
It's kinda awful to have a /dev/console device, that just returns ENODEV if you try to use it... I#d understand if ttynull would make it just eat all data, like /dev/null does it. But returning ENODEV? that's just mean, they should have called it console=enodev or so.

**Comment by poettering** [comment]
So regarding the first issue i filed this now: #31269, because I figure we should handle ENODEV graceuflly for synchronizing VT access.

**Comment by poettering** [comment]
for the other issue, not sure it's worth addressing. 

I mean, we could in theory add StandardInput=tty-graceful or so, which is just like StandardInput=tty but falls back to /dev/null in case /dev/console is not accessible.

**Comment by Winterhuman** [comment]
Yeah, I don't mind seeing that warning myself, I more just opened this issue so this weird issue is known and documented here. Feel free to close this issue if adding `StandardInput=tty-graceful` won't be necessary

**Comment by Winterhuman** [comment]
I've found an easier way of detecting the NULL TTY driver. When it's active, `/sys/class/tty/console/active` doesn't exist, so a simple `ConditionPathExists=` line will work in that case. I'll send a PR for adding that to `systemd-firstboot.service`
